cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -291,6 +296,23 @@ private static String getMetadata(String resource) throws IOException {
     }
   }
 
+  /**
+   * Recursively deletes all the contents of the directory and the directory itself.
+   */
+  public static void deleteDirectoryContents(File file) {
```

Comment:
We probably need to guard against potential exception if we call delete concurrently.

Reply:
Currently provisioners are initialized sequentially. So we do need to worry about concurrency here?

Reply:
Let's not put this method in this class as it deletes local directory and has nothing to do with Dataproc.

Reply:
check for file being null.